cdap-docs/admin-manual/source/security/authorization.rst

```diff
@@ -304,90 +303,155 @@ Streams
+     - *READ*
    * - Sending events to a stream (sync, async, or batch)
-     - *WRITE* (on the stream) and *READ* (on the namespace)
+     - *WRITE*
    * - Drop
      - *ADMIN*
    * - Drop-all in the namespace
-     - *ADMIN* (on the namespace)
+     - *ADMIN* on all streams in the namespace
    * - Update
      - *ADMIN*
    * - Truncate
      - *ADMIN*
-   * - List
-     - Only returns those artifacts on which user has at least one of *READ, WRITE, EXECUTE,* or *ADMIN*
-   * - View
+   * - List/View
+     - Only returns those streams on which user has at least one of *READ, WRITE, EXECUTE,* or *ADMIN*
+   * - Get
      - At least one of *READ, WRITE, EXECUTE,* or *ADMIN*
 
+.. _security-authorization-policies-principal:
 
-.. _security-bootstrapping-authorization:
+Principal
+---------
 
-Bootstrapping Authorization
-===========================
-When CDAP is first started with authorization enabled, no users are granted privileges on
-any CDAP entities. Without any privileges, it can be impossible to bootstrap CDAP (create
-a new namespace or create entities in the *default* namespace) unless an
-external interface (such as `Hue <http://gethue.com/>`__) is used for a supported
-authorization extension (such as :ref:`Integrations: Apache Sentry <apache-sentry>`).
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
 
-To make this bootstrap process easier, during startup the CDAP Master issues these grants to
-select users:
+   * - Operation
+     - Privileges Required
+   * - Deploy an app to impersonate a principal
+     - *ADMIN* on the principal
+   * - Create a namespace with owner prinicpal
+     - *ADMIN* on the principal
+   * - Create a dataset with owner prinicpal
+     - *ADMIN* on the principal
+   * - Create a stream with owner prinicpal
+     - *ADMIN* on the principal
 
-- The user that runs the CDAP Master is granted *ADMIN* on the CDAP instance, so that the
-  *default* namespace can be created by that user if it does not already exist.
 
-- The user that runs the CDAP Master is granted *READ*, *WRITE*, *EXECUTE*, and
-  *ADMIN* on the system namespace, so operations such as creation of system tables,
-  deployment of system artifacts, and deployment of system dataset modules can be
-  performed by that user.
+.. _security-pre-grant-wildcard-privilege:
+
+Pre-grant and WildCard Privileges
+=================================
+The new authorization model requires pre-granted *ADMIN* privilege on all entity creation.
+When CDAP is first started with authorization enabled, no users are granted privileges on
+any CDAP entities. Without any privileges, CDAP will not be able to create the default namespace.
+To create the default namespace, grant *ADMIN* on default namespace to the CDAP master user.
+The default namespace will get created in several minutes.
+
+To pre-grant the privilege, wildcard policy is supported. ``*`` can be used to represent any character and words. ``?``
```

Comment:
The CLI example is given in the sentry integration part since I think it is different for each extension. For ranger we are using UI to grant privileges. So we can add example for ranger once we have it realeased.